sw/airborne/modules/imu/imu.c

```diff
@@ -529,6 +561,8 @@ void imu_init(void)
     if(!imu.mags[i].calibrated.scale) {
       VECT3_ASSIGN(imu.mags[i].scale[0], IMU_MAG_X_SIGN, IMU_MAG_Y_SIGN, IMU_MAG_Z_SIGN);
       VECT3_ASSIGN(imu.mags[i].scale[1], 1, 1, 1);
+    }if(!imu.mags[i].calibrated.scale_f) {
+      VECT3_ASSIGN(imu.mags[i].scale_f, IMU_MAG_X_SIGN, IMU_MAG_Y_SIGN, IMU_MAG_Z_SIGN);
```

Comment:
We can check here if `scale` is set instead and calculate `scale_f`

Reply:
That's a very good idea !
I'm on it.